fv3gfs-util/fv3gfs/util/partitioner.py

```diff
@@ -141,18 +158,79 @@ def global_extent(self, rank_metadata: QuantityMetadata) -> Tuple[int, ...]:
             rank_metadata.dims, rank_metadata.extent, self.layout
         )
 
-    def subtile_extent(self, global_metadata: QuantityMetadata) -> Tuple[int, ...]:
-        """Return the shape of a single rank representation for the given dimensions."""
-        return rank_extent_from_tile_metadata(
-            global_metadata.dims, global_metadata.extent, self.layout
+    def subtile_extent(
+        self,
+        global_metadata: QuantityMetadata,
+        rank: int,
+        is_cube_metadata: bool = False,
+    ) -> Tuple[int, ...]:
+        """Return the shape of a single rank representation for the given dimensions.
+
+        Args:
+            global_metadata: quantity metadata.
+            rank: rank of the process.
+            is_cube_metadata (optional): True if the given metadata is for a cubed sphere model.
+                False if not, e.g. data is for a single tile. Default is False.
+
+        Returns:
+            extent: shape of a single rank representation for the given dimensions.
+        """
+        if is_cube_metadata:
+            cube_metadata_offset = 1
+        else:
+            cube_metadata_offset = 0
+
+        # cube_metadata_offset is used to exclude the TILE dimension (first index) for cubed sphere metadata.
+        tile_decomposition = subtile_extents_from_tile_metadata(
+            global_metadata.dims[cube_metadata_offset:],
+            global_metadata.extent[cube_metadata_offset:],
+            self.layout,
         )
```

Comment:
The tile dimension is not necessarily guaranteed to be the first dimension, but it is guaranteed to have the value `pace.util.TILE_DIM`. Can you remove the `is_cube_metadata` argument and detect the presence of (and remove) the TILE_DIM instead?

Reply:
`is_cube_metadata` has been removed and replaced by a check for a dimension that matches `util.constants.TILE_DIM`.
This check detects the TILE_DIM in any location within the given dimensions, which means that the assertion of it needing to be in the first index can potentially be removed. This assertion is located in `CubedSpherePartitioner.subtile_extent`

```
        if cube_metadata.dims[0] != constants.TILE_DIM:
            raise NotImplementedError(
                "currently only supports tile dimension {constants.TILE_DIM} as the "
                "first dimension, got dims {cube_metadata.dims}"
            )
```